### PR TITLE
rewards call

### PR DESCRIPTION
**Pull request by MaxMillington**
## What is the purpose of the change

<!-- > Add a description of the overall background and high level changes that this PR introduces

This is the code for collecting staking rewards

### ClickUp Task

[ClickUp Task URL](https://app.clickup.com/t/8678m13z8)

## Testing and Verifying

<!-- _(Please pick either of the following options)_

This change has been tested locally by rebuilding the website and verified content and links are expected

### CODE REVIEW

File: packages/stores/src/account/osmosis/types.ts [hunk 1]
```diff
@@ -129,6 +129,12 @@ export const osmosisMsgOpts = createMsgOpts({
       osmosis.valsetpref.v1beta1.MessageComposer.withTypeUrl
         .delegateToValidatorSet,
   },
+  withdrawDelegationRewards: {
```
Comment (MaxMillington) [inline packages/stores/src/account/osmosis/types.ts:132]: reran telescop. We still get the wrong type from the proto. Told Sishir

**Comment by MaxMillington** [comment]
@jonator Any insight here, I'm getting a test failure in the stores package:

```
src/account/osmosis/types.ts:118:10 - error TS2339: Property 'lockExistingFullRangePositionAndSFStake' does not exist on type '{ superfluidDelegate(value: MsgSuperfluidDelegate): { typeUrl: string; value: MsgSuperfluidDelegate; }; superfluidUndelegate(value: MsgSuperfluidUndelegate): { ...; }; ... 6 more ...; addToConcentratedLiquiditySuperfluidPosition(value: MsgAddToConcentratedLiquiditySuperfluidPosition): { ...; }; }'.

118         .lockExistingFullRangePositionAndSFStake,
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

**Comment by jonator** [comment]
> 

https://github.com/osmosis-labs/osmosis-frontend/pull/2012/files#diff-1ca7a8b77f7536e0a20aaccb2e8c762cbc58ae9ac32d4d6ee1102fab9e3b7301L118

I believe that message was removed. The fix for this is in this PR ^^

File: packages/web/components/cards/rewards-card.tsx [hunk 2]
```diff
@@ -8,9 +8,13 @@ export const RewardsCard: React.FC<{
...
   tooltipContent: string;
-}> = ({ title, titleIconUrl, tooltipContent }) => {
+  onClick: () => void;
+}> = ({ title, titleIconUrl, tooltipContent, onClick }) => {
   return (
-    <div className="flex w-full flex-grow flex-col rounded-xl border-2 border-osmoverse-600">
+    <div
+      className="flex w-full flex-grow cursor-pointer flex-col rounded-xl border-2 border-osmoverse-600"
+      onClick={onClick}
+    >
```
Comment (jonator) [inline packages/web/components/cards/rewards-card.tsx:17]: It appears this is a redundantly nested div? Let me know what that is for, perhaps for the border?

File: packages/web/components/cards/stake-dashboard.tsx [hunk 3]
```diff
@@ -77,11 +87,13 @@ export const StakeDashboard: React.FC<{
             title={t("stake.collectRewards")}
             titleIconUrl="www.google.com"
             tooltipContent="... placeholder content 1 ..."
+            onClick={collectRewards}
           />
           <RewardsCard
             title={t("stake.investRewards")}
             titleIconUrl="www.google.com"
             tooltipContent="... placeholder content 2 ..."
```
Comment (jonator) [inline packages/web/components/cards/stake-dashboard.tsx:95]: While we're here, could we remove the google URL and placeholder content so it doesn't get forgotten?

**Review by jonator — APPROVED** [review]

**Pull request merged**